### ISSUE TITLE
IC8 is reported as failed on +2A/+3 machines using Diagboard hardware.

### ISSUE THREAD

**Issue by brendanalford** [brendanalford/zx-diagnostics#29]
zx-diagnostics is consistently reporting IC8 as failed when running the ROM test on +2A/+3 machines using Ben Versteeg's diagboard implementation. 

The code works without error on other diagnostic hardware (e.g. Retroleum SMART card).

**Comment by brendanalford** [comment]
Okay, at this point I believe this to be a hardware issue.

On a 48/128/grey +2, there is a single /ROMCS signal that controls whether the internal ROM is inhibited or not. The +2AB/+3, since they have two physical ROM chips, now have two independent edge connector signals to inhibit each ROM - /ROMOE1 and /ROMOE2.

There is onboard circuitry on the diagnostic board to detect whether or not the host system is a +2A/+3 or not, and this is based on sensing whether the 9v edge connector line (not connected on a +2A/+3) is present and thus using a potential divider to drive an analogue switch to allow the /ROMOE2 line to be driven (this must be disconnected on 48/128/+2 machines as this position on the edge connector is the composite video output on these machines).

The issue seems to be that the signal for /ROMOE2 isn't being driven low enough when requested for the second ROM (IC8) to be enabled when requested, hence the second part of the ROM test fails.

The problem also manifests itself when setting J1 to 2-3 (Spectrum mode), on my +2A at least the system behaves as if the onboard ROM isn't there as /ROMCS is held high, and you get the familiar white/blue vertical striped screen.

ZX-diagnostics' ROM detection and test code works fine on the SMART card as the CPLD seems to handle the detection and drive of these signals correctly.

In short, there's nothing that can be done from a software point of view to work around this issue, since from software a single bit in the card's I/O register (OUT 31) controls whether the diagnostic board's ROM is paged in or not - it does not control the individual ROMs.

**Comment by brendanalford** [comment]
This was subsequently confirmed to be an issue with the newer Diagboard hardware. A PCB patch was made available which resolved the issue such that zx-diagnostics no longer considers IC8 faulty when it is not.
